scripts/build.js

```diff
@@ -1,86 +1,73 @@
-    'dev',
+    'production',
+    'development',
 ];
 
-/**
- * Creates an object of output options for rollup.
- * @param {string} directory - the component's folder.
- * @param {string} format - the module type to which to bundle.
- * @param {string} moduleName - the name of the bundle.
- * @param {boolean} isProd - if true, the code will be minified.
- * @returns {object} - OutputParams object.
-*/
-function generateOutputOptions(directory, format = 'umd', moduleName, isProd = false) {
-    const suffix = isProd ? '.production.min' : '.development';
-
-    const outputOptions = {
-        format: format,
-        dir: path.join(directory, format),
-        entryFileNames: `${moduleName}${suffix}.js`,
-        globals: {
-            'coherent-gameface-components': 'components',
-        },
-        exports: 'auto',
-    };
-
-    // When bundling for umd we need to specify a name for
-    // the global variable that will be exported.
-    if (format === 'umd') outputOptions.name = moduleName;
-    // terser is a rollup plugin that minifies the code
-    if (isProd) outputOptions.plugins = [terser(), strip()];
-
-    return outputOptions;
-}
-
 /**
  * Creates bundles for given list of formats and environments.
  * @param {string} moduleName - the root name of the bundle.
  * @param {object} inputOptions - rollup input options.
- * @param {Array<string>} formats - the module types for which to bundle(UMD, CJS).
  * @param {Array<string>} environments - the environments for which to bundle(prod, dev).
  * @param {string} libPath - the path to either the library or a component.
 */
-async function buildAndPackage(moduleName, inputOptions, formats, environments, libPath) {
-    for (const format of formats) {
-        for (const environment of environments) {
-            await createBundle(inputOptions, generateOutputOptions(
-                path.dirname(inputOptions.input),
-                format,
-                moduleName,
-                environment === 'prod' ? true : false
-            ));
-        }
+async function buildAndPackage(moduleName, inputOptions, environments, libPath) {
+    for (const environment of environments) {
+        const suffix = environment === 'development' ? '.development' : '.production.min';
+        const name = `${moduleName}${suffix}.js`;
+
+        const config = baseConfig(environment);
+        let additionalOutputConfig = {};
+
+        if (moduleName === 'components') additionalOutputConfig = libraryConfig.output;
+
+        await buildWithWebpack({
+            ...config,
+            entry: inputOptions.entry,
+            output: {
+                path: path.join(path.dirname(inputOptions.entry), 'dist'),
+                filename: name,
+                ...additionalOutputConfig,
+            },
+        }).catch(console.error);
+
+        // npm pack must run after createBundle has finished, otherwise the rollup
+        // bundling will not be ready yet and not everything will be packed.

```

Comment:
```suggestion
        // npm pack must run after createBundle has finished, otherwise the webpack
        // bundling will not be ready yet and not everything will be packed.
```